pallets/stableswap/src/lib.rs

```diff
@@ -313,6 +327,8 @@ pub mod pallet {
 		) -> DispatchResult {
 			T::AuthorityOrigin::ensure_origin(origin)?;
 
+			let amplification = NonZeroU16::new(amplification).ok_or(Error::<T>::InvalidAmplification)?;
+
 			let pool_id = Self::do_create_pool(share_asset, &assets, amplification, trade_fee, withdraw_fee)?;
```

Comment:
should emit the event `AmplificationChanging` for consistency

Reply:
i dont think this is necessary. It creates pool with given amplification. It is not going to change.

Reply:
yeah, but if you hook on events to get fees, you will miss the initial setting

Reply:
not like we are creating that many pools so we cannot afford extra event

Reply:
added